pkg/server/ca/manger_health.go

```diff
@@ -0,0 +1,50 @@
+package ca
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/spiffe/spire/pkg/common/health"
+)
+
+// TODO: What would be a good threshold number?
+const failedRotationThreshold = 10
+
+type managerHealth struct {
+	m *Manager
+}
+
+func (h *managerHealth) CheckHealth() health.State {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	err := h.m.rotate(ctx)
+
+	// For liveness, we'll check if the rotations are done sucessfully.
+	live := err == nil
+
+	// Readiness will be checked by manager's ability to rotate
+	// for a certain threshold.
+	ready := true
+	var readyErr error
+	if h.m.failedRotationNum > failedRotationThreshold {
```

Comment:
Accessing the count directly from a separate goroutine that the one updating it is not safe. We should instead add a method to the Manager to get at the rotation failure counter that protects access to the counter via a lock or atomics. The bumping of the counter inside of manager should also use either a lock or atomics to protect concurrent access.

Reply:
Thanks. Learned `sync/atomic` today!